src/engines/fx/chorus/screen.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+
+#include "core/ui/vector_graphics.hpp"
+#include <Gamma/Envelope.h>
+#include "chorus.hpp"
+
+
+
+namespace otto::engines::chorus {
+
+  using namespace core;
+
+  struct Screen : ui::Screen {
+    void draw(nvg::Canvas& ctx) override;
+    void draw_front_head(nvg::Canvas&, nvg::Point, ui::vg::Color, float);
+    void draw_background_head(nvg::Canvas&, nvg::Point, ui::vg::Color, float);
+
+    float delay_ = 0.f;
+    void action(itc::prop_change<&Props::delay>, float d) noexcept;
+    float depth_ = 0.f;
+    void action(itc::prop_change<&Props::depth>, float d) noexcept;
+    float feedback_ = 0.f;
+    void action(itc::prop_change<&Props::feedback>, float f) noexcept;
+    float rate_ = 0.f;
+    void action(itc::prop_change<&Props::rate>, float r) noexcept;
+    std::atomic<float>* phase_ = nullptr;
+    void action(Actions::phase_value, std::atomic<float>& v) noexcept;
+
+    std::array<ui::vg::Color, 10> colour_list = {ui::vg::Colors::Blue,
```

Comment:
this array should be constexpr and declared in the .cpp file. `vg::Color` is `constexpr`-enabled on `develop` now, so this should be doable if you pull from that.